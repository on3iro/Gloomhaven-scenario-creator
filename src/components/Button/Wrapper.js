/**
 * Styled container for Button
 *
 * @namespace Button.Wrapper
 * @memberOf Button
  */

import styled from 'styled-components';


const Wrapper = styled.div`
  display: inline-block;
`;

export default Wrapper;
