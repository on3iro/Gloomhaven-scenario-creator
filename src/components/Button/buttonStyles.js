/**
 * Creates button specific styles which can be applied to different elements/components
 *
 * @namespace Button.buttonStyles
 * @memberOf Button
  */

import { css } from 'styled-components';


const buttonStyles = css`
  display: inline-block;
  text-decoration: none;
  padding: 5px 10px;
  background-color: ${props => props.warning ? 'red' : '#27a70b'};
  color: #fff;
  outline: 0;
  border-radius: 4px;
  font-weight: bold;

  &:hover {
    background-color: ${props => props.warning ? 'tomato' : '#49ab33'};
  }
`;

export default buttonStyles;
