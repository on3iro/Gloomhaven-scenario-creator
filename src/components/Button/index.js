/**
 * A common button, if you pass it a prop "route" it'll render a link to a react-router route.
 * Otherwise it'll render a link with an onclick.
 *
 * @namespace Button
  */

import React, { PropTypes, Children } from 'react';

import A from './A';
import StyledButton from './StyledButton';
import StyledSubmit from './StyledSubmit';
import Wrapper from './Wrapper';


const Button = props => {
  // Render an anchor tag
  let button = (
    <A href={props.href} onClick={props.onClick} {...props} >
      {Children.toArray(props.children)}
    </A>
  );

  // If the button has a handleRoute prop, we want to render a button
  if(props.handleRoute) {
    button = (
      <StyledButton onClick={props.handleRoute} {...props} >
        {Children.toArray(props.children)}
      </StyledButton>
    );
  }

  // If the button has submit prop render input
  if(props.submit) {
    button = (
      <StyledSubmit {...props} />
    );
  }

  return (
    <Wrapper>
      {button}
    </Wrapper>
  );
};

Button.propTypes = {
  handleRoute: PropTypes.func,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  submit: PropTypes.bool,
};

export default Button;
