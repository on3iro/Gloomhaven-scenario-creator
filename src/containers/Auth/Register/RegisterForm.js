/**
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React, { Component } from 'react';

import H1 from 'components/H1';
import Button from 'components/Button';
import Input from 'components/Input';


class RegisterForm extends Component {
  /**
    * Class component RegisterForm
    *
    * @class RegisterForm
    * @namespace RegisterForm
    * @memberOf RegisterPage.RegisterForm
    */

  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      confirm: '',
    };
  }

  handleInputChange = (event) => {
    /**
      * Handles changes on input elements.
      * Sets the value of the respective input.
      *
      * @method handleInputChange
      * @memberOf RegisterPage.RegisterForm.RegisterForm
      * @namespace handleInputChange
      * @param {Object} event - Triggered event
      * @returns {undefined}
      */

    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });

    return undefined;
  }

  render() {
    return (
      <form onSubmit={() => {}}>
        <H1>Register</H1>
        <Input
          name="email"
          type="text"
          placeholder="Email"
          value={this.state.email}
          onChange={this.handleInputChange}
        />
        <Input
          name="password"
          type="password"
          placeholder="Password"
          value={this.state.password}
          onChange={this.handleInputChange}
        />
        <Input
          name="confirm"
          type="password"
          placeholder="Confirm Password"
          value={this.state.confirm}
          onChange={this.handleInputChange}
        />
        <Button submit>Register Now</Button>
      </form>
    );
  }
}

export default RegisterForm;
