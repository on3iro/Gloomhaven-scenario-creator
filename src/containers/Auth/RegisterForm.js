/*
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';

import H1 from 'components/H1';
import Button from 'components/Button';

import RenderField from './RenderField';


export const validate = values => {
  const errors = {};
  const REQUIRED = 'Required';

  if(!values.email) {
    errors.email = REQUIRED;
  }else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }

  if(!values.password) {
    errors.password = REQUIRED;
  }

  if(!values.name) {
    errors.name = REQUIRED;
  }

  if(!values.confirm) {
    errors.confirm = REQUIRED;
  }else if(values.confirm !== values.password) {
    errors.confirm = 'Passwords do not match';
  }

  return errors;
};

export const RegisterForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Register</H1>
      <Field name="email" type="text"
        component={RenderField} label="email"
      />
      <Field name="name" type="text"
        component={RenderField} label="name"
      />
      <Field name="password" type="password"
        component={RenderField} label="password"
      />
      <Field name="confirm" type="password"
        component={RenderField} label="confirm"
      />
      <Button submit>Register Now</Button>
    </form>
  );
};

RegisterForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'register',
  validate
})(RegisterForm);
