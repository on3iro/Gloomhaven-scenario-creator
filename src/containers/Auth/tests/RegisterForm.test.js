import React from 'react';
import { shallow } from 'enzyme';

import H1 from 'components/H1';
import Button from 'components/Button';
import { Field } from 'redux-form';
import { RegisterForm } from '../RegisterForm';


const renderComponent = newProps => {
  const props = {
    handleSubmit: jest.fn(),
  };

  const renderedWrapper = shallow(<RegisterForm {...props} />);

  return {
    props,
    renderedWrapper,
  };
};

describe('<RegisterForm />', () => {
  it('should render <form>', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find('form').length).toMatchSnapshot();
  });

  it('should render <H1>', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(H1).length).toMatchSnapshot();
  });

  it('should render Fields', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Field).length).toMatchSnapshot();
  });

  it('Should render <Button />', () => {
    const { renderedWrapper } = renderComponent();
    expect(renderedWrapper.find(Button).length).toMatchSnapshot();
  });
});
