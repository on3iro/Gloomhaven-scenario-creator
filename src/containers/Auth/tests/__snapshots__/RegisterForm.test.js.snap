exports[`<RegisterForm /> Should render <Button /> 1`] = `1`;

exports[`<RegisterForm /> should render <H1> 1`] = `1`;

exports[`<RegisterForm /> should render <form> 1`] = `1`;

exports[`<RegisterForm /> should render Fields 1`] = `4`;

exports[`validate() should throw "Invald email address" 1`] = `
Object {
  "email": "Invalid email address",
}
`;

exports[`validate() should throw "Passwords do not match" 1`] = `
Object {
  "confirm": "Passwords do not match",
}
`;

exports[`validate() should throw "Required" on confirm 1`] = `
Object {
  "confirm": "Required",
}
`;

exports[`validate() should throw "Required" on empty email 1`] = `
Object {
  "email": "Required",
}
`;

exports[`validate() should throw "Required" on empty name 1`] = `
Object {
  "name": "Required",
}
`;

exports[`validate() should throw "Required" on password 1`] = `
Object {
  "confirm": "Passwords do not match",
  "password": "Required",
}
`;
